bundles/org.openhab.binding.enigma2/src/main/java/org/openhab/binding/enigma2/handler/Enigma2Handler.java

```diff
@@ -0,0 +1,304 @@
+ *
+ * @author Guido Dolfen - Initial contribution
+ */
+@NonNullByDefault
+public class Enigma2Handler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(Enigma2Handler.class);
+    private Enigma2Configuration configuration = new Enigma2Configuration();
+    private Optional<Enigma2Client> enigma2Client = Optional.empty();
+    @Nullable
+    private ScheduledFuture<?> refreshJob;
+    private LocalDateTime lastAnswerTime = LocalDateTime.now();
+
+    public Enigma2Handler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        configuration = getConfigAs(Enigma2Configuration.class);
+        if (configuration.host.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "host must not be empty");
+        } else if (configuration.timeout <= 0 || configuration.timeout > 300) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "timeout must be between 0 and 300 seconds");
+        } else if (configuration.refreshInterval <= 0 || configuration.refreshInterval > 3600) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "refreshInterval must be between 0 and 3600 seconds");
+        }
+        enigma2Client = Optional.of(new Enigma2Client(configuration.host, configuration.user, configuration.password,
+                configuration.timeout));
+        refreshJob = scheduler.scheduleWithFixedDelay(this::refresh, 2, configuration.refreshInterval, TimeUnit.SECONDS);
+    }
+
+    private void refresh() {
+        getEnigma2Client().ifPresent(client -> {
+            boolean online = client.refresh();
+            if (online) {
+                updateStatus(ThingStatus.ONLINE);
+                updateState(CHANNEL_POWER, client.isPower() ? OnOffType.ON : OnOffType.OFF);
+                updateState(CHANNEL_MUTE, client.isMute() ? OnOffType.ON : OnOffType.OFF);
+                updateState(CHANNEL_VOLUME, new PercentType(client.getVolume()));
+                updateState(CHANNEL_CHANNEL, new StringType(client.getChannel()));
+                updateState(CHANNEL_TITLE, new StringType(client.getTitle()));
+                updateState(CHANNEL_DESCRIPTION, new StringType(client.getDescription()));
+                if (lastAnswerTime.isBefore(client.getLastAnswerTime())) {
+                    lastAnswerTime = client.getLastAnswerTime();
+                    updateState(CHANNEL_ANSWER, new StringType(client.getAnswer()));
+                }
+            } else {
+                updateStatus(ThingStatus.OFFLINE);
+            }
+        });
+    }
+
+    @Override
+    @SuppressWarnings("null")
+    public void dispose() {
+        if(refreshJob != null) {
+            if (!refreshJob.isCancelled()) {
+                refreshJob.cancel(true);
+            }
+        }
+        refreshJob = null;
```

Comment:
Just to deal with the null check warnings. Also, no harm in cancelling something twice.
```suggestion
        ScheduledFuture<?> refreshJob = this.refreshJob;
        if(refreshJob != null) {
            refreshJob.cancel(true);
        }
        this.refreshJob = null;
```

Reply:
Fixed